tests/robot-tests/tests/admin/bau/upload_files.robot

```diff
@@ -48,7 +48,7 @@ Upload a ZIP file subject
     user checks headed table body row contains    Data file    absence_in_prus.csv    ${section}
     user checks headed table body row contains    Metadata file    absence_in_prus.meta.csv    ${section}
     user checks headed table body row contains    Data file size    141 Kb    ${section}
-    user checks headed table body row contains    Number of rows    613    ${section}
+    user checks headed table body row contains    Number of rows    612    ${section}
```

Comment:
How come all of these row counts have shrunk by 1?

Reply:
If you uploaded a non-zip data file in the Admin prior to this change, or uploaded a data zip and the importer extracted a data file from a zip archive, this dictionary was created and set on the blob metadata of the data file blob:

```
var dataInfo = GetDataFileMetaValues(
    metaFileName: metaFile.Filename,
    numberOfRows: CalculateNumberOfRows(dataFormFile.OpenReadStream())
);
```

`CalculateNumberOfRows` read the number of _lines_ in the file.

The same number was written to the `DataImport.Rows` field in the db which has now been renamed `TotalRows`.

Calculating the number of rows is now always deferred to 'Stage 1' of the importer so the Admin app is not potentially iterating over a huge number of rows just to count them.

DataImport.TotalRows is now set after validating the data file  in 'Stage 1' and it's the number of rows in the `DataRowCollection` object from the CSV, which doesn't include the header row.

Now 'TotalRows' and 'ImportedRows' should match each unless any rows are excluded from the import.